### PR TITLE
runtime, envelope: add support for `cgroup` for `child` processes.

### PR DESCRIPTION
**Pull request by flouthoc**
* Allow envelope to accept cgroup manager which gives users opportunity to
constraint process created by weavelets, which is really helpful in case
of `multi` and `single` deployment where user would want to limit `cpu`,
`memory` usage or other resource usage.

* multi: create and provide cgroup manager

### CODE REVIEW

File: internal/tool/multi/deployer.go [hunk 1]
```diff
@@ -109,6 +111,33 @@ type handler struct {
...
+			Mems:   "0",
+		},
+	}
+	cgroupManager, err := cgroups.NewManager("/sys/fs/cgroup", groupPath, &res)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create a new cgroup: %w", err)
+	}
+	return cgroupManager, nil
+}
+
```
Comment (flouthoc) [inline internal/tool/multi/deployer.go:140]: This is just a stub, if approved in final PR i will refactor this bit to show up from `config` files.

**Comment by flouthoc** [comment]
@rgrandl @mwhittaker While working on one of my project I was needed to spawn weavelets under fixed CPU and memory constraints so I added support for that in my private fork, I think this will be a handy feature for upstream. Do you have any opinions on this one ?

Ofcourse I need to harden this PR a bit more so cgroup specs can be accepted from `config` but I wanted to see if maintainers like this feature or not.

File: internal/tool/multi/deployer.go [hunk 2]
```diff
@@ -109,6 +111,33 @@ type handler struct {
...
```
Comment (flouthoc) [inline internal/tool/multi/deployer.go:140]: This is just a stub, if feature is approved by maintainers then I will amend this PR so we read actual cgroup parameters from config file.

**Comment by flouthoc** [comment]
@rgrandl just a gentle bump to see if you were able to get time to discuss this one :)

**Comment by rgrandl** [comment]
@flouthoc, we've discussed this. 

We were thinking of an alternate solution: would it be possible to create these cgroups beforehand, and then when you start the child process, specify the name of the cgroup(s) under which the process should run?

**Comment by flouthoc** [comment]
@rgrandl Yes sounds reasonable to me, so service-weaver should only contain the part of moving processes to the cgroup, is that correct and user should specify `cgroup` name via config ? 

Also Just curious any specific reason about why `cgroup` created by the service-weaver is not in the vision ?

**Comment by flouthoc** [comment]
@rgrandl Done, added change so now it can accept externally created `cgroup` via config. I have only wired `multi` deployer for now so its easier to review.

Let me know if config fields look fine then I can also add this feature to `single` deployer or maybe for `single` deployer I can open a new PR later.

**Comment by flouthoc** [comment]
> Let me know if config fields look fine then I can also add this feature to single deployer or maybe for single deployer I can open a new PR later.

@rgrandl Any thoughts on this ?

**Comment by flouthoc** [comment]
Gentle bump on the PR.

**Comment by flouthoc** [comment]
> > Let me know if config fields look fine then I can also add this feature to single deployer or maybe for single deployer I can open a new PR later.
> 
> @rgrandl Any thoughts on this ?

Gentle bump on the PR.

File: internal/tool/multi/deployer.go (hunk 2 above)
Comment (flouthoc) [reply internal/tool/multi/deployer.go]: This is resolved.

**Comment by flouthoc** [comment]
Wondering if anything is pending on this PR from my side. I am just waiting for another review round on this one.

**Comment by rgrandl** [comment]
@flouthoc , sorry for being quiet on this. I've been distracted with some other things. I've been waiting to discuss your change with some other team members, because the design is a little bit intrusive/tricky, but haven't had a chance to do that yet. 

Is this a blocker for you? I will come back with comments on this soon.

**Comment by rgrandl** [comment]
This is something we've never managed to agree on as a team. Close it for now.

**Comment by flouthoc** [comment]
Sure no issues :)

**Pull request closed without merging**